src/server_manager/model/account.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ManagedServer, RegionId} from "./server";
+
+// Keys are cityIds like "nyc".  Values are regions like ["nyc1", "nyc3"].
+export type RegionMap = {
+  [cityId: string]: RegionId[]
+};
+
+// Repository of ManagedServer objects.  These servers are created by the server
+// manager on cloud providers where we can provide a "magical" user experience,
+// e.g. DigitalOcean.
+export interface Account {
```

Comment:
The generic account should have name, id and status only for now, to be used in the the side bar and status check.

RegionMap, listServers, getRegionMap and createServer are all DO-specific and don't belong here.

My suggestion is to move DigitalOceanServer and DigitalOceanAccount to a `model/digitalocean.ts` as Server and Account. This way you can refer to them as `do.Server` and `do.Account`.

The restriction to be in model/ is that this file must only depend on infrastructure/ and model/. There must be no dependency on the app. (infrascruture can only depend on infrastructure)

Reply:
The alternative here is to have DigitalOcean{Account,Server} interfaces in the model, and then the implementations (with same name?) kept in the app.
That may be better if we need to inject DO objects in tests